ogre/src/OgreLidarVisual.cc

```diff
@@ -0,0 +1,301 @@
+ * 
+ */
+
+#include <ignition/common/Console.hh>
+#include "ignition/rendering/ogre/OgreDynamicLines.hh"
+#include "ignition/rendering/ogre/OgreLidarVisual.hh"
+#include "ignition/rendering/ogre/OgreScene.hh"
+#include "ignition/rendering/ogre/OgreMarker.hh"
+#include "ignition/rendering/ogre/OgreGeometry.hh"
+
+
+class ignition::rendering::OgreLidarVisualPrivate
+{
+  /// \brief Non Hitting DynamicLines Object to display
+  public: std::vector<std::shared_ptr<OgreDynamicLines>> noHitRayStrips;
+
+  /// \brief Hitting DynamicLines Object to display
+  public: std::vector<std::shared_ptr<OgreDynamicLines>> rayStrips;
+
+  /// \brief Dead Zone Geometry DynamicLines Object to display
+  public: std::vector<std::shared_ptr<OgreDynamicLines>> deadZoneRayFans;
+
+  /// \brief Lidar Ray DynamicLines Object to display
+  public: std::vector<std::shared_ptr<OgreDynamicLines>> rayLines;
+
+  /// \brief The current lidar data message
+  public: std::vector<double> lidarPoints;
+
+  /// \brief True if new message is received
+  public: bool receivedMsg;
+};
+
+using namespace ignition;
+using namespace rendering;
+
+//////////////////////////////////////////////////
+OgreLidarVisual::OgreLidarVisual()
+  : dataPtr(new OgreLidarVisualPrivate)
+{
+}
+
+//////////////////////////////////////////////////
+OgreLidarVisual::~OgreLidarVisual()
+{
+  this->Destroy();
+}
+
+//////////////////////////////////////////////////
+void OgreLidarVisual::PreRender()
+{
+  // no ops
+}
+
+//////////////////////////////////////////////////
+void OgreLidarVisual::Destroy()
+{
+  for (auto ray : this->dataPtr->noHitRayStrips)
+  {
+    ray->Clear();
+    ray.reset();
+  }
+
+  for (auto ray : this->dataPtr->rayStrips)
+  {
+    ray->Clear();
+    ray.reset();
+  }
+
+  for (auto ray : this->dataPtr->rayLines)
+  {
+    ray->Clear();
+    ray.reset();
+  }
+
+  for (auto ray : this->dataPtr->deadZoneRayFans)
+  {
+    ray->Clear();
+    ray.reset();
+  }
+
+  this->ClearPoints();
+}
+
+//////////////////////////////////////////////////
+void OgreLidarVisual::Init()
+{
+  BaseLidarVisual::Init();
+  this->Create();
+}
+
+//////////////////////////////////////////////////
+void OgreLidarVisual::Create()
+{
+  this->ClearPoints();
+  this->dataPtr->receivedMsg = false;
+}
+
+//////////////////////////////////////////////////
+void OgreLidarVisual::ClearPoints()
+{
+  this->dataPtr->noHitRayStrips.clear();
+  this->dataPtr->deadZoneRayFans.clear();
+  this->dataPtr->rayLines.clear();
+  this->dataPtr->rayStrips.clear();
+}
+
+
+//////////////////////////////////////////////////
+void OgreLidarVisual::SetPoints(const std::vector<double> &_msg)
+{
+  this->dataPtr->lidarPoints = _msg;
+  this->dataPtr->receivedMsg = true;
+}
+
+//////////////////////////////////////////////////
+void OgreLidarVisual::Update()
+{
+  if (!this->dataPtr->receivedMsg || this->dataPtr->lidarPoints.size() == 0)
+  {
+    ignerr << "Message not received. Exiting update function"
```

Comment:
can you change this to `ignwarn` and update the message to say something like: `Empty lidar points. Exiting update function`?